From 7dba8b4bfba92b7a4e6c5190a881fcd591ee2360 Mon Sep 17 00:00:00 2001
From: Ladislav Kunc <ladislavkunc@gmail.com>
Date: Tue, 12 Mar 2013 20:04:03 +0100
Subject: [PATCH] Data connection detach after a timeout period fix

Some network carriers will detach data connection after a timeout period,
isDataAllowed() will not allow a data connection because
mAutoAttachOnCreation is false and gprsState is not STATE_IN_SERVICE. This
bug was fixed in Android 4.2.2. This ports the fix (Change-Id:
If41c2514ebdb9cd2fba88eef692f2c7160c04b0f) to Android 4.1.2

Bug: http://code.google.com/p/freexperia/issues/detail?id=1390
Change-Id: Ibc31718df512356ff3313fa4a35ce74174c34789
---
 .../com/android/internal/telephony/gsm/GsmDataConnectionTracker.java  |    1 +
 1 file changed, 1 insertion(+)

diff --git a/telephony/java/com/android/internal/telephony/gsm/GsmDataConnectionTracker.java b/telephony/java/com/android/internal/telephony/gsm/GsmDataConnectionTracker.java
index 6d47d03..8f64733 100644
--- a/telephony/java/com/android/internal/telephony/gsm/GsmDataConnectionTracker.java
+++ b/telephony/java/com/android/internal/telephony/gsm/GsmDataConnectionTracker.java
@@ -626,6 +626,7 @@ public final class GsmDataConnectionTracker extends DataConnectionTracker {
             notifyOffApnsOfAvailability(Phone.REASON_DATA_ATTACHED);
         }
 
+        mAutoAttachOnCreation = true;
         setupDataOnReadyApns(Phone.REASON_DATA_ATTACHED);
     }
 
-- 
1.7.10.4

